scripts/build-images-json.py

```diff
@@ -0,0 +1,46 @@
+import json
+import os
+import sys
+
+def load_config(config_path):
+    """Load JSON configuration from the given path."""
+    with open(config_path, 'r') as file:
+        return json.load(file)
+
+def main(image_list_file):
+    # Load the list of image directories
+    with open(image_list_file, 'r') as file:
+        image_dirs = json.load(file)
+
+    print("Image directories:", image_dirs)  
+
+    # Load the default build config
+    default_config_path = 'opensciencegrid/default-build-config.json'
+    default_config = load_config(default_config_path)
+
+    image_matrix = {}
+
+    for image_dir in image_dirs:
+        if image_dir == "dummy":
+            continue
+
+        build_config_path = os.path.join(image_dir, 'build-config.json')
+
+        if os.path.exists(build_config_path):
+            config = load_config(build_config_path)
+        else:
+            config = default_config
```

Comment:
In Python, the best practice is to `ask for forgiveness rather than permission` as this helps avoid race conditions (what happens if the build config disappears between when you check it and when you load it?). Use a `try` / `except` block instead